pkg/backend/storage/fs/store.go

```diff
@@ -54,7 +55,25 @@ func (s *Store) Set(ctx context.Context, name string, value []byte) error {
 func (s *Store) Delete(ctx context.Context, name string) error {
 	path := filepath.Join(s.path, filepath.Clean(name))
 	out.Debug(ctx, "fs.Delete(%s) - %s", name, path)
-	return os.Remove(path)
+
+	err := os.Remove(path)
+	if err != nil {
+		return err
+	}
+
+	// remove all empty parent directories
+	for dir := filepath.Dir(path); ; dir = filepath.Dir(dir) {
+		err = os.Remove(dir)
+		switch {
+		case err == nil:
+			continue
+		case err.(*os.PathError).Err == syscall.ENOTEMPTY:
+			// ignore when directory is non-empty
+			return nil
+		default:
+			return err
+		}
```

Comment:
This loop makes me feel uneasy. This reminds me of that one time when I identified a function annotated with "recursive delete, can go up" for deleting a whole production system. While this shouldn't happen I would prefer to limit it strictly to never exceed the password store root dir.

Reply:
You are absolutely right. Even though it _should_ not happen, the effects of this action must remain limited to the gopass managed directory.